src/Compiler/Compiler.cpp

```diff
@@ -1461,33 +1461,248 @@ bool Compiler::RegisterCommands(TclInterpreter* interp, bool batchMode) {
+                       "can specify its time units. Ex: 1ps, 1000ns, 1s"),
+            nullptr);
+        return TCL_ERROR;
+      }
+
+      std::string cmd = std::string("gtkwave::setWindowStartTime ") + argv[1];
+
+      // Send cmd to GTKWave
+      compiler->GTKWaveSendCmd(cmd);
+    }
+    return TCL_OK;
+  };
+  interp->registerCmd("wave_time", wave_time, this, nullptr);
+
+  auto wave_show = [](void* clientData, Tcl_Interp* interp, int argc,
+                      const char* argv[]) -> int {
+    Compiler* compiler = (Compiler*)clientData;
+    if (compiler) {
+      if (argc != 2) {
+        Tcl_AppendResult(interp,
+                         qPrintable("Expected Syntax: wave_show <signalName>"),
+                         nullptr);
+        return TCL_ERROR;
+      }
+      std::string cmd = "foedag::show_signal " + std::string(argv[1]);
+      compiler->GTKWaveSendCmd(cmd);
+    }
 
     return TCL_OK;
   };
-  interp->registerCmd("gtkwave", gtkwave, this, nullptr);
+  interp->registerCmd("wave_show", wave_show, this, nullptr);
+
+  auto wave_refresh = [](void* clientData, Tcl_Interp* interp, int argc,
+                         const char* argv[]) -> int {
+    Compiler* compiler = (Compiler*)clientData;
+    if (compiler) {
+      compiler->GTKWaveSendCmd("gtkwave::reloadFile");
+    }
+    return TCL_OK;
+  };
+  interp->registerCmd("wave_refresh", wave_refresh, this, nullptr);
 
   return true;
 }
 
+// This will send a given command to the gtkwave wish interface over stdin
+void Compiler::GTKWaveSendCmd(const std::string& gtkWaveCmd,
+                              bool raiseGtkWindow /* true */) {
+  QProcess* process = GetGTKWaveProcess();
+
+  std::string cmd = gtkWaveCmd;
+  if (process && process->isWritable()) {
+    if (cmd.back() != '\n') {
+      // Add a newline to end of command if one is missing
+      // (GTKWave will hang if you don't send a newline with your command)
+      cmd += "\n";
+    }
+
+    if (raiseGtkWindow) {
+      // Add a window raise command before the command to try to bring gtkwave
+      // to focus
+      cmd = "gtkwave::presentWindow\n" + cmd;
+    }
+
+    // Send command to GTKWave's stdin
+    process->write(cmd.c_str());
+    process->waitForBytesWritten();
+  }
+}
+
+// This will return a pointer to the current gtkwave process, if no process is
+// running, one will be started
+QProcess* Compiler::GetGTKWaveProcess() {
+  if (!m_gtkwave_process) {
+    m_gtkwave_process = new QProcess();
+
+    // Get a path to our local gtkwave binary
+    auto binPath = GlobalSession->Context()->BinaryPath();
+    auto exePath = binPath / "gtkwave" / "bin";
+    QString wd = QString::fromStdString(exePath.string());
+    // Need to be in the gtkwave bin dir to resolve some dependency issues
+    m_gtkwave_process->setWorkingDirectory(wd);
+
+    // Enable process control via tcl interface over std in with --wish
+    QString wishArg = "--wish";
```

Comment:
Same stdd:string question here.

Reply:
QProcess is Qt and it uses a QStringList which wishArg gets stored in so we could use std::string for a line, but we are going to have to use QString eventually...